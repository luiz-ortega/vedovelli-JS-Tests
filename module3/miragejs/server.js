import { Server } from "miragejs";

const products = [
  {
    title: "RelÃ³gio bonito",
    price: "22.00",
    image:
      "https://images.unsplash.com/photo-1495856458515-0637185db551?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80",
  },
];

export const makeServer = ({ environment = "development" } = {}) => {
  return new Server({
    environment,
    routes() {
      this.namespace = "api";
      this.get("products", () => ({
        products,
      }));
    },
  });
};
