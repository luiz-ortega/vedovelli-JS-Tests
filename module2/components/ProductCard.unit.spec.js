import { mount } from '@vue/test-utils';
import ProductCard from '@/components/ProductCard';
import { makeServer } from '@/miragejs/server';

describe('ProductCard - unit', () => {
  const mountProductCard = () => {
    const product = server.create('product', {
      title: 'Relógio bonito',
      price: '24.00',
      image:
        'https://images.unsplash.com/photo-1495856458515-0637185db551?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80',
    });

    return {
      wrapper: mount(ProductCard, {
        propsData: {
          product,
        },
      }),
      product,
    };
  };

  let server;
  beforeEach(() => {
    server = makeServer({ environment: 'test' });
  });

  afterEach(() => {
    server.shutdown();
  });

  it('should match snapshot', () => {
    const wrapper = mountProductCard();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should mount the component', () => {
    const { wrapper } = mountProductCard();

    expect(wrapper.vm).toBeDefined();
    expect(wrapper.text()).toContain('Relógio bonito');
    expect(wrapper.text()).toContain('$24.00');
  });

  it('should emit the event addToCart with product object when button gets clicked', async () => {
    const { wrapper, product } = mountProductCard();

    await wrapper.find('button').trigger('click');

    expect(wrapper.emitted().addToCart).toBeTruthy();
    expect(wrapper.emitted().addToCart.length).toBe(1);
    expect(wrapper.emitted().addToCart[0]).toEqual([{ product }]);
  });
});
