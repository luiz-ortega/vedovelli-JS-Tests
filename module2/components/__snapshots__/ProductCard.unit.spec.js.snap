// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard - unit should match snapshot 1`] = `undefined`;
